app/main.py

```diff
@@ -1,3 +1,66 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Any:
+        if not isinstance(other, Distance):
+            other = Distance(other)
+        return Distance(km=self.km + other.km)
+
+    def __radd__(self, other: Any) -> Any:
+        if other == 0:
+            return self
+        else:
+            return self.__add__(other)
+
+    def __iadd__(self, other: Any) -> Any:
+        if not isinstance(other, Distance):
+            other = Distance(other)
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: Any) -> Any:
+        if isinstance(other, Distance):
+            raise TypeError
+        return Distance(km=self.km * other)
+
+    def __truediv__(self, other: Any) -> Any:
+        if isinstance(other, Distance):
+            raise TypeError
```

Comment:
It's recommended to raise a `TypeError` with a descriptive message, e.g., `raise TypeError('Cannot divide by a Distance object')`.